update to use openssl 1.x dependency

M-Waszkiewicz-Anaconda opened the pull request:
create another variant with openssl 1.x dependency
- change version name +openssl1
- use arrow-cpp +openssl1 dependency variant 
- use snowflake channel
-

skupr-anaconda commented:
`conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform linux-64: {'arrow-cpp==10.0.1+openssl1'}`

skupr-anaconda commented:
Not sure why but arrow-cpp 10.0.1+openssl1 artifacts have been created only for aarch64.
Looking at the logs of this PR I see clashes of dependencies. I guess openssl 3.x and 1.x meet in the same environment

lorepirri commented:
Also the last commit does not have all the builds:
https://staging.continuum.io/prefect/fs/arrow-cpp-feedstock/pr34/878f03d

skupr-anaconda commented:
Definitely openssl 1.x and 3.x in the same invironment:
`"openssl[version='>=1.1.1j,<1.1.2a|>=1.1.1k,<1.1.2a|>=1.1.1l,<1.1.2a|>=1.1.1n,<1.1.2a|>=1.1.1q,<1.1.2a|>=1.1.1s,<1.1.2a|>=1.1.1t,<1.1.2a|>=1.1.1u,<1.1.2a|>=1.1.1v,<1.1.2a|>=3.0.10,<4.0a0|>=3.0.9,<4.0a0|>=3.0.8,<4.0a0']"`

M-Waszkiewicz-Anaconda commented:
Tests done:
``
base) mwaszkiewicz@DESKTOP-75NFDUK:~/aggregate$ conda create -n test_snowflake_connector -c local openssl=1.1.1 
Collecting package metadata (current_repodata.json): done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 23.7.4
  latest version: 23.9.0

Please update conda by running

    $ conda update -n base -c defaults conda

Or to minimize the number of packages updated during conda update use

     conda install conda=23.9.0



## Package Plan ##

  environment location: /home/mwaszkiewicz/miniconda3/envs/test_snowflake_connector

  added / updated specs:
    - openssl=1.1.1


The following NEW packages will be INSTALLED:

  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main 
  _openmp_mutex      pkgs/main/linux-64::_openmp_mutex-5.1-1_gnu 
  ca-certificates    pkgs/main/linux-64::ca-certificates-2023.08.22-h06a4308_0 
  libgcc-ng          pkgs/main/linux-64::libgcc-ng-11.2.0-h1234567_1 
  libgomp            pkgs/main/linux-64::libgomp-11.2.0-h1234567_1 
  openssl            pkgs/main/linux-64::openssl-1.1.1w-h7f8727e_0 


Proceed ([y]/n)? y


Downloading and Extracting Packages

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate test_snowflake_connector
#
# To deactivate an active environment, use
#
#     $ conda deactivate

(base) mwaszkiewicz@DESKTOP-75NFDUK:~/aggregate$ conda activate test_snowflake_connector
(test_snowflake_connector) mwaszkiewicz@DESKTOP-75NFDUK:~/aggregate$ conda install snowflake-connector-python=3.2
Collecting package metadata (current_repodata.json): done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 23.7.4
  latest version: 23.9.0

Please update conda by running

    $ conda update -n base -c defaults conda

Or to minimize the number of packages updated during conda update use

     conda install conda=23.9.0



## Package Plan ##

  environment location: /home/mwaszkiewicz/miniconda3/envs/test_snowflake_connector

  added / updated specs:
    - snowflake-connector-python=3.2


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    arrow-cpp-10.0.1           |       h374c478_1        10.0 MB
    snowflake-connector-python-3.2.0|  py311ha02d727_1         947 KB
    ------------------------------------------------------------
                                           Total:        10.9 MB

The following NEW packages will be INSTALLED:

  abseil-cpp         pkgs/main/linux-64::abseil-cpp-20211102.0-hd4dd3e8_0 
  arrow-cpp          pkgs/main/linux-64::arrow-cpp-10.0.1-h374c478_1 
  asn1crypto         pkgs/main/linux-64::asn1crypto-1.5.1-py311h06a4308_0 
  aws-c-common       pkgs/main/linux-64::aws-c-common-0.6.8-h5eee18b_1 
  aws-c-event-stream pkgs/main/linux-64::aws-c-event-stream-0.1.6-h6a678d5_6 
  aws-checksums      pkgs/main/linux-64::aws-checksums-0.1.11-h5eee18b_2 
  aws-sdk-cpp        pkgs/main/linux-64::aws-sdk-cpp-1.8.185-h721c034_1 
  blas               pkgs/main/linux-64::blas-1.0-mkl 
  boost-cpp          pkgs/main/linux-64::boost-cpp-1.82.0-hdb19cb5_2 
  brotlipy           pkgs/main/linux-64::brotlipy-0.7.0-py311h5eee18b_1002 
  bzip2              pkgs/main/linux-64::bzip2-1.0.8-h7b6447c_0 
  c-ares             pkgs/main/linux-64::c-ares-1.19.1-h5eee18b_0 
  certifi            pkgs/main/linux-64::certifi-2023.7.22-py311h06a4308_0 
  cffi               pkgs/main/linux-64::cffi-1.15.1-py311h5eee18b_3 
  charset-normalizer pkgs/main/noarch::charset-normalizer-2.0.4-pyhd3eb1b0_0 
  cryptography       pkgs/main/linux-64::cryptography-41.0.3-py311hdda0065_0 
  filelock           pkgs/main/linux-64::filelock-3.9.0-py311h06a4308_0 
  gflags             pkgs/main/linux-64::gflags-2.2.2-he6710b0_0 
  glog               pkgs/main/linux-64::glog-0.5.0-h2531618_0 
  gmp                pkgs/main/linux-64::gmp-6.2.1-h295c915_3 
  grpc-cpp           pkgs/main/linux-64::grpc-cpp-1.48.2-he1ff14a_1 
  icu                pkgs/main/linux-64::icu-73.1-h6a678d5_0 
  idna               pkgs/main/linux-64::idna-3.4-py311h06a4308_0 
  intel-openmp       pkgs/main/linux-64::intel-openmp-2023.1.0-hdb19cb5_46305 
  krb5               pkgs/main/linux-64::krb5-1.20.1-h143b758_1 
  ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.38-h1181459_1 
  libboost           pkgs/main/linux-64::libboost-1.82.0-h109eef0_2 
  libbrotlicommon    pkgs/main/linux-64::libbrotlicommon-1.0.9-h5eee18b_7 
  libbrotlidec       pkgs/main/linux-64::libbrotlidec-1.0.9-h5eee18b_7 
  libbrotlienc       pkgs/main/linux-64::libbrotlienc-1.0.9-h5eee18b_7 
  libcurl            pkgs/main/linux-64::libcurl-8.4.0-h251f7ec_0 
  libedit            pkgs/main/linux-64::libedit-3.1.20221030-h5eee18b_0 
  libev              pkgs/main/linux-64::libev-4.33-h7f8727e_1 
  libevent           pkgs/main/linux-64::libevent-2.1.12-hdbd6064_1 
  libffi             pkgs/main/linux-64::libffi-3.4.4-h6a678d5_0 
  libnghttp2         pkgs/main/linux-64::libnghttp2-1.57.0-h2d74bed_0 
  libprotobuf        pkgs/main/linux-64::libprotobuf-3.20.3-he621ea3_0 
  libssh2            pkgs/main/linux-64::libssh2-1.10.0-hdbd6064_2 
  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-11.2.0-h1234567_1 
  libthrift          pkgs/main/linux-64::libthrift-0.15.0-h1795dd8_2 
  libuuid            pkgs/main/linux-64::libuuid-1.41.5-h5eee18b_0 
  lz4-c              pkgs/main/linux-64::lz4-c-1.9.4-h6a678d5_0 
  mkl                pkgs/main/linux-64::mkl-2023.1.0-h213fc3f_46343 
  mkl-service        pkgs/main/linux-64::mkl-service-2.4.0-py311h5eee18b_1 
  mkl_fft            pkgs/main/linux-64::mkl_fft-1.3.8-py311h5eee18b_0 
  mkl_random         pkgs/main/linux-64::mkl_random-1.2.4-py311hdb19cb5_0 
  ncurses            pkgs/main/linux-64::ncurses-6.4-h6a678d5_0 
  numpy              pkgs/main/linux-64::numpy-1.26.0-py311h08b1b3b_0 
  numpy-base         pkgs/main/linux-64::numpy-base-1.26.0-py311hf175353_0 
  orc                pkgs/main/linux-64::orc-1.7.4-hb3bc3d3_1 
  oscrypto           pkgs/main/noarch::oscrypto-1.2.1-pyhd3eb1b0_0 
  packaging          pkgs/main/linux-64::packaging-23.1-py311h06a4308_0 
  pip                pkgs/main/linux-64::pip-23.2.1-py311h06a4308_0 
  platformdirs       pkgs/main/linux-64::platformdirs-3.8.1-py311h06a4308_0 
  pyarrow            pkgs/main/linux-64::pyarrow-10.0.1-py311hd8e8d9b_0 
  pycparser          pkgs/main/noarch::pycparser-2.21-pyhd3eb1b0_0 
  pycryptodomex      pkgs/main/linux-64::pycryptodomex-3.15.0-py311h5eee18b_0 
  pyjwt              pkgs/main/linux-64::pyjwt-2.4.0-py311h06a4308_0 
  pyopenssl          pkgs/main/linux-64::pyopenssl-23.2.0-py311h06a4308_0 
  pysocks            pkgs/main/linux-64::pysocks-1.7.1-py311h06a4308_0 
  python             pkgs/main/linux-64::python-3.11.5-h955ad1f_0 
  pytz               pkgs/main/linux-64::pytz-2023.3.post1-py311h06a4308_0 
  re2                pkgs/main/linux-64::re2-2022.04.01-h295c915_0 
  readline           pkgs/main/linux-64::readline-8.2-h5eee18b_0 
  requests           pkgs/main/linux-64::requests-2.31.0-py311h06a4308_0 
  setuptools         pkgs/main/linux-64::setuptools-68.0.0-py311h06a4308_0 
  snappy             pkgs/main/linux-64::snappy-1.1.9-h295c915_0 
  snowflake-connect~ pkgs/main/linux-64::snowflake-connector-python-3.2.0-py311ha02d727_1 
  sortedcontainers   pkgs/main/noarch::sortedcontainers-2.4.0-pyhd3eb1b0_0 
  sqlite             pkgs/main/linux-64::sqlite-3.41.2-h5eee18b_0 
  tbb                pkgs/main/linux-64::tbb-2021.8.0-hdb19cb5_0 
  tk                 pkgs/main/linux-64::tk-8.6.12-h1ccaba5_0 
  tomlkit            pkgs/main/linux-64::tomlkit-0.11.1-py311h06a4308_0 
  typing-extensions  pkgs/main/linux-64::typing-extensions-4.7.1-py311h06a4308_0 
  typing_extensions  pkgs/main/linux-64::typing_extensions-4.7.1-py311h06a4308_0 
  tzdata             pkgs/main/noarch::tzdata-2023c-h04d1e81_0 
  urllib3            pkgs/main/linux-64::urllib3-1.26.16-py311h06a4308_0 
  utf8proc           pkgs/main/linux-64::utf8proc-2.6.1-h27cfd23_0 
  wheel              pkgs/main/linux-64::wheel-0.41.2-py311h06a4308_0 
  xz                 pkgs/main/linux-64::xz-5.4.2-h5eee18b_0 
  zlib               pkgs/main/linux-64::zlib-1.2.13-h5eee18b_0 
  zstd               pkgs/main/linux-64::zstd-1.5.5-hc292b87_0 

The following packages will be UPDATED:

  openssl                                 1.1.1w-h7f8727e_0 --> 3.0.11-h7f8727e_2 


Proceed ([y]/n)? y


Downloading and Extracting Packages
                                                                                                                                                                        
Preparing transaction: done                                                                                                                                             
Verifying transaction: done
Executing transaction: done
``

skupr-anaconda commented:
Is it correct that openssl will be updated to 3.0.11?:
```
The following packages will be UPDATED:

  openssl                                 1.1.1w-h7f8727e_0 --> 3.0.11-h7f8727e_2 


Proceed ([y]/n)? y
```

M-Waszkiewicz-Anaconda commented:
yes, @skupr-anaconda  you are right it wasn't good test, see this instead:
```

(base) mwaszkiewicz@DESKTOP-75NFDUK:~/aggregate$ conda create -c https://staging.continuum.io/prefect/fs/arrow-cpp-feedstock/pr34/0ed9e35 -c https://staging.continuum.io/prefect/fs/arrow-cpp-feedstock/pr34/0ed9e35  -n test_snowflake_
connector --dry-run openssl=1 snowflake-connector-python=3.2
Collecting package metadata (current_repodata.json): done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 23.7.4
  latest version: 23.9.0

Please update conda by running

    $ conda update -n base -c defaults conda

Or to minimize the number of packages updated during conda update use

     conda install conda=23.9.0



## Package Plan ##

  environment location: /home/mwaszkiewicz/miniconda3/envs/test_snowflake_connector

  added / updated specs:
    - openssl=1
    - snowflake-connector-python=3.2


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    arrow-cpp-10.0.1           |       h35c149b_2        10.0 MB  https://staging.continuum.io/prefect/fs/arrow-cpp-feedstock/pr34/0ed9e35
    ------------------------------------------------------------
                                           Total:        10.0 MB

The following NEW packages will be INSTALLED:

  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main 
  _openmp_mutex      pkgs/main/linux-64::_openmp_mutex-5.1-1_gnu 
  abseil-cpp         pkgs/main/linux-64::abseil-cpp-20211102.0-hd4dd3e8_0 
  arrow-cpp          prefect/fs/arrow-cpp-feedstock/pr34/0ed9e35/linux-64::arrow-cpp-10.0.1-h35c149b_2 
  asn1crypto         pkgs/main/linux-64::asn1crypto-1.5.1-py311h06a4308_0 
  aws-c-common       pkgs/main/linux-64::aws-c-common-0.4.57-he6710b0_1 
  aws-c-event-stream pkgs/main/linux-64::aws-c-event-stream-0.1.6-h2531618_5 
  aws-checksums      pkgs/main/linux-64::aws-checksums-0.1.9-he6710b0_0 
  aws-sdk-cpp        pkgs/main/linux-64::aws-sdk-cpp-1.8.185-hce553d0_0 
  blas               pkgs/main/linux-64::blas-1.0-mkl 
  boost-cpp          pkgs/main/linux-64::boost-cpp-1.82.0-hdb19cb5_2 
  brotlipy           pkgs/main/linux-64::brotlipy-0.7.0-py311h5eee18b_1002 
  bzip2              pkgs/main/linux-64::bzip2-1.0.8-h7b6447c_0 
  c-ares             pkgs/main/linux-64::c-ares-1.19.1-h5eee18b_0 
  ca-certificates    pkgs/main/linux-64::ca-certificates-2023.08.22-h06a4308_0 
  certifi            pkgs/main/linux-64::certifi-2023.7.22-py311h06a4308_0 
  cffi               pkgs/main/linux-64::cffi-1.15.1-py311h5eee18b_3 
  charset-normalizer pkgs/main/noarch::charset-normalizer-2.0.4-pyhd3eb1b0_0 
  cryptography       pkgs/main/linux-64::cryptography-41.0.3-py311h130f0dd_0 
  filelock           pkgs/main/linux-64::filelock-3.9.0-py311h06a4308_0 
  gflags             pkgs/main/linux-64::gflags-2.2.2-he6710b0_0 
  glog               pkgs/main/linux-64::glog-0.5.0-h2531618_0 
  gmp                pkgs/main/linux-64::gmp-6.2.1-h295c915_3 
  grpc-cpp           pkgs/main/linux-64::grpc-cpp-1.48.2-h5bf31a4_0 
  icu                pkgs/main/linux-64::icu-73.1-h6a678d5_0 
  idna               pkgs/main/linux-64::idna-3.4-py311h06a4308_0 
  intel-openmp       pkgs/main/linux-64::intel-openmp-2023.1.0-hdb19cb5_46305 
  krb5               pkgs/main/linux-64::krb5-1.20.1-h568e23c_1 
  ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.38-h1181459_1 
  libboost           pkgs/main/linux-64::libboost-1.82.0-h109eef0_2 
  libbrotlicommon    pkgs/main/linux-64::libbrotlicommon-1.0.9-h5eee18b_7 
  libbrotlidec       pkgs/main/linux-64::libbrotlidec-1.0.9-h5eee18b_7 
  libbrotlienc       pkgs/main/linux-64::libbrotlienc-1.0.9-h5eee18b_7 
  libcurl            pkgs/main/linux-64::libcurl-8.2.1-h91b91d3_0 
  libedit            pkgs/main/linux-64::libedit-3.1.20221030-h5eee18b_0 
  libev              pkgs/main/linux-64::libev-4.33-h7f8727e_1 
  libevent           pkgs/main/linux-64::libevent-2.1.12-h8f2d780_0 
  libffi             pkgs/main/linux-64::libffi-3.4.4-h6a678d5_0 
  libgcc-ng          pkgs/main/linux-64::libgcc-ng-11.2.0-h1234567_1 
  libgomp            pkgs/main/linux-64::libgomp-11.2.0-h1234567_1 
  libnghttp2         pkgs/main/linux-64::libnghttp2-1.52.0-ha637b67_1 
  libprotobuf        pkgs/main/linux-64::libprotobuf-3.20.3-he621ea3_0 
  libssh2            pkgs/main/linux-64::libssh2-1.10.0-h37d81fd_2 
  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-11.2.0-h1234567_1 
  libthrift          pkgs/main/linux-64::libthrift-0.15.0-h0d84882_2 
  libuuid            pkgs/main/linux-64::libuuid-1.41.5-h5eee18b_0 
  lz4-c              pkgs/main/linux-64::lz4-c-1.9.4-h6a678d5_0 
  mkl                pkgs/main/linux-64::mkl-2023.1.0-h213fc3f_46343 
  mkl-service        pkgs/main/linux-64::mkl-service-2.4.0-py311h5eee18b_1 
  mkl_fft            pkgs/main/linux-64::mkl_fft-1.3.8-py311h5eee18b_0 
  mkl_random         pkgs/main/linux-64::mkl_random-1.2.4-py311hdb19cb5_0 
  ncurses            pkgs/main/linux-64::ncurses-6.4-h6a678d5_0 
  numpy              pkgs/main/linux-64::numpy-1.26.0-py311h08b1b3b_0 
  numpy-base         pkgs/main/linux-64::numpy-base-1.26.0-py311hf175353_0 
  openssl            pkgs/main/linux-64::openssl-1.1.1w-h7f8727e_0 
  orc                pkgs/main/linux-64::orc-1.7.4-hb3bc3d3_1 
  oscrypto           pkgs/main/noarch::oscrypto-1.2.1-pyhd3eb1b0_0 
  packaging          pkgs/main/linux-64::packaging-23.1-py311h06a4308_0 
  pip                pkgs/main/linux-64::pip-23.2.1-py311h06a4308_0 
  platformdirs       pkgs/main/linux-64::platformdirs-3.8.1-py311h06a4308_0 
  pyarrow            pkgs/main/linux-64::pyarrow-10.0.1-py311hd8e8d9b_0 
  pycparser          pkgs/main/noarch::pycparser-2.21-pyhd3eb1b0_0 
  pycryptodomex      pkgs/main/linux-64::pycryptodomex-3.15.0-py311h5eee18b_0 
  pyjwt              pkgs/main/linux-64::pyjwt-2.4.0-py311h06a4308_0 
  pyopenssl          pkgs/main/linux-64::pyopenssl-23.2.0-py311h06a4308_0 
  pysocks            pkgs/main/linux-64::pysocks-1.7.1-py311h06a4308_0 
  python             pkgs/main/linux-64::python-3.11.5-h7a1cb2a_0 
  pytz               pkgs/main/linux-64::pytz-2023.3.post1-py311h06a4308_0 
  re2                pkgs/main/linux-64::re2-2022.04.01-h295c915_0 
  readline           pkgs/main/linux-64::readline-8.2-h5eee18b_0 
  requests           pkgs/main/linux-64::requests-2.31.0-py311h06a4308_0 
  setuptools         pkgs/main/linux-64::setuptools-68.0.0-py311h06a4308_0 
  snappy             pkgs/main/linux-64::snappy-1.1.9-h295c915_0 
  snowflake-connect~ pkgs/main/linux-64::snowflake-connector-python-3.2.0-py311ha02d727_1 
  sortedcontainers   pkgs/main/noarch::sortedcontainers-2.4.0-pyhd3eb1b0_0 
  sqlite             pkgs/main/linux-64::sqlite-3.41.2-h5eee18b_0 
  tbb                pkgs/main/linux-64::tbb-2021.8.0-hdb19cb5_0 
  tk                 pkgs/main/linux-64::tk-8.6.12-h1ccaba5_0 
  tomlkit            pkgs/main/linux-64::tomlkit-0.11.1-py311h06a4308_0 
  typing-extensions  pkgs/main/linux-64::typing-extensions-4.7.1-py311h06a4308_0 
  typing_extensions  pkgs/main/linux-64::typing_extensions-4.7.1-py311h06a4308_0 
  tzdata             pkgs/main/noarch::tzdata-2023c-h04d1e81_0 
  urllib3            pkgs/main/linux-64::urllib3-1.26.16-py311h06a4308_0 
  utf8proc           pkgs/main/linux-64::utf8proc-2.6.1-h27cfd23_0 
  wheel              pkgs/main/linux-64::wheel-0.41.2-py311h06a4308_0 
  xz                 pkgs/main/linux-64::xz-5.4.2-h5eee18b_0 
  zlib               pkgs/main/linux-64::zlib-1.2.13-h5eee18b_0 
  zstd               pkgs/main/linux-64::zstd-1.5.5-hc292b87_0 




```

M-Waszkiewicz-Anaconda commented:
> I'm looking at the logs and the test environment has `openssl: 3.0.11-h7f8727e_2 defaults`. Is there something missing?

 could you elaborate on this ?

M-Waszkiewicz-Anaconda commented:
> > > I'm looking at the logs and the test environment has `openssl: 3.0.11-h7f8727e_2 defaults`. Is there something missing?
> > 
> > 
> > could you elaborate on this ?
> 
> It doesn't look like you need a rebuild of this package. The existing one should install with the openssl 1 version, now that you have a compatible arrow-cpp.

yes, but this is specially crafted to go to snowflake channel only, as on defaults we do not want openssl 1
